src/library/dao/playlistdao.cpp

```diff
@@ -980,18 +982,16 @@ void PlaylistDAO::shuffleTracks(const int playlistId, const QList<int>& position
         trackPositionIds.insert(trackBPosition, trackAId);
         newPositions.swap(newPositions.indexOf(trackAPosition),
                           newPositions.indexOf(trackBPosition));
-        QString swapQuery = "UPDATE PlaylistTracks SET position=%1 "
-                "WHERE position=%2 AND playlist_id=%3";
-        query.exec(swapQuery.arg(QString::number(-1),
-                                 QString::number(trackAPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackAPosition),
-                                 QString::number(trackBPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackBPosition),
-                                 QString::number(-1),
-                                 QString::number(playlistId)));
 
+        qSwap(playListTrackIds[trackAPosition - 1],
+              playListTrackIds[trackBPosition - 1]);
```

Comment:
What should I do now, I think this algo is fast as the changing of the list in the database happens only 1 time and no repeated shuffling takes place and in memory shuffling takes place which is faster and if we assume that there are 10^6 songs(assuming 1mb song each) then it would be reasonably fast with O(n) complexity so it is fast I think